crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -424,13 +425,22 @@ impl UserIdentityData {
 ///
 /// The user can explicitly pin the new identity to allow for legitimate
 /// identity changes (for example, in case of key material or device loss).
+///
+/// As soon as the cryptographic identity is verified (i.e. signed by our own
+/// trusted identity), a latch is set to remember it (`verified_latch`). Future
+/// interactions will expect this user to stay verified, in case of violation
+/// the user should be notified with a blocking warning.
 #[derive(Debug, Clone, Deserialize, Serialize)]
 #[serde(try_from = "OtherUserIdentityDataSerializer", into = "OtherUserIdentityDataSerializer")]
 pub struct OtherUserIdentityData {
     user_id: OwnedUserId,
     pub(crate) master_key: Arc<MasterPubkey>,
     self_signing_key: Arc<SelfSigningPubkey>,
     pinned_master_key: Arc<RwLock<MasterPubkey>>,
+    /// This tracks whether we've ever verified this user with any identity at
+    /// any point in time on this device. To use it in the future to detect
+    /// cases where the user has become unverified for any reason.
+    verified_latch: Arc<AtomicBool>,
```

Comment:
Why `_latch`? Does it have a specific meaning here that is helpful?

I would have expected just `verified` maybe.

Reply:
Hmm, reading later commits, maybe `was_verified` would be a good name?

Reply:
I found the `latch`(electronic) interesting because it convey that it stays like that until you actively reset it.
In opposition to, only the previous identity was verified. So maybe `was_verified_once`?

Reply:
`previously_verified` is the closest I can get. If you like `latch` I can live with it.

Reply:
Err, let's not use latch, nobody will understand this without reading the doc comment, and then they'll be decently confused why not a more straight forward name was picked.

Reply:
Using `previously_verified` as per suggestion